java/client/src/org/openqa/selenium/remote/AddRotatable.java

```diff
@@ -19,28 +19,61 @@
 
 import com.google.common.collect.ImmutableMap;
 
+import org.openqa.selenium.DeviceRotation;
 import org.openqa.selenium.Rotatable;
 import org.openqa.selenium.ScreenOrientation;
 
 import java.lang.reflect.Method;
 
 public class AddRotatable implements AugmenterProvider {
 
+  private enum RotatableMethod {
+    ROTATE("rotate"),
+    GET_ORIENTATION("getOrientation"),
+    GET_ROTATION("rotation");
+      
+    private String methodName;
+      
+    private RotatableMethod(String methodName) {
+      this.methodName = methodName;
+    }
+    
+    public String toString() {
+      return this.methodName;
+    }
+  }
+    
   public Class<?> getDescribedInterface() {
     return Rotatable.class;
   }
 
   public InterfaceImplementation getImplementation(Object value) {
     return new InterfaceImplementation() {
       public Object invoke(ExecuteMethod executeMethod, Object self, Method method, Object... args) {
-        if ("rotate".equals(method.getName())) {
-          return executeMethod.execute(DriverCommand.SET_SCREEN_ORIENTATION,
-              ImmutableMap.of("orientation", args[0]));
-        } else if ("getOrientation".equals(method.getName())) {
-          return ScreenOrientation.valueOf((String) executeMethod.execute(
-              DriverCommand.GET_SCREEN_ORIENTATION, null));
+        RotatableMethod m = RotatableMethod.valueOf(method.getName());
```

Comment:
you don't need to create a enum for this, java allows you to switch on a string ;) (since java 7?)